### PR TITLE
Configuration improvements

### PR DESCRIPTION
**Pull request by altro3**
Improvements in classes structure:

1. Move configuration methods to ConfigUtils class
2. Move file methods to FileUtils class
3. Added autosave to context already loaded config value
4. Replace System.getProperty(...) to context.getOptions().get(...) for supported annotaton processor options

### CODE REVIEW

**Comment by altro3** [comment]
@graemerocher Hi! Now all as you want: I use context.getOpetions.get() instead System.getProperty()

File: openapi/src/main/java/io/micronaut/openapi/visitor/ConfigProperty.java [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Supported configuration properties.
+ *
+ * @since 4.10.0
+ */
+public interface ConfigProperty {
```
Comment (graemerocher) [inline openapi/src/main/java/io/micronaut/openapi/visitor/ConfigProperty.java:27]: I would rename this to something less generic like `OpenApiConfigProperty` or something

**Review by graemerocher — APPROVED** [review]

File: openapi/src/main/java/io/micronaut/openapi/visitor/ConfigProperty.java (hunk 1 above)
Comment (altro3) [reply openapi/src/main/java/io/micronaut/openapi/visitor/ConfigProperty.java]: @graemerocher fixed

**Comment by altro3** [comment]
@graemerocher sorry, I need to revert `context.getOptions().get(...)` to `System.getProperty(...)` because of it https://github.com/micronaut-projects/micronaut-core/issues/9671

**Pull request merged**